### ISSUE TITLE
Spice Of Life Crashing The Game

### ISSUE THREAD

**Issue by Crystal98TR** [GTNewHorizons/GT-New-Horizons-Modpack#10267]
### Your GTNH Discord Username

Crystal98#3352

### Your Pack Version

2.1.2.3

### Your Server

SP

### Type of Server

Single Player

### Your Actions

I was literally just walking around, then I tried to open the Lunch Box's inventory, and the game freezes followed with a crash. Read the report for a lot more info.

### Crash Report

[crash-2022-05-03_16.56.21-client.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/8617286/crash-2022-05-03_16.56.21-client.txt)

After the initial crash, I can't even join in the world anymore, it instantly freezes and crashes.

However I can join in using a backup, even the one that's created right before the crash.

Throwing the Lunch Box out of my inventory and never using it doesn't help. Because not only using the item crashes the game, but also trying to check random recipes with NEI and even trying to open a quest with a food reward inside.

I can consistently crash the game by using the same save/backup files. Not sure if playing in a completely different world doesn't crash the game, as I didn't test any of the stuff above including opening the Lunch Box thing, but I can at least join in the world unlike the main one where you can't even join after the first crash.

Deleting the Spice Of Life: Carrot Edition fixes it and everything works completely fine. Deleting and adding it back doesn't.

I downloaded the classic Spice Of Life instead and it also works fine.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by Chaoschaot232** [comment]
Same for another player on my mp server: https://pastebin.com/tHsWTqYs

Not sure what is missing or what causes this problem generally, you are not alone with this problem :)

**Comment by Crystal98TR** [comment]
Happy (or sad lol) to hear I'm not alone! _SpiceOfLife-1.7.10-v.2.0.0-carrot-beta10.jar_ is the name of the file so it's beta I guess? I didn't change anything though, it came with the modpacks itself.

**Comment by Chaoschaot232** [comment]
Yes, same version as ours. Try out to get https://github.com/GTNewHorizons/SpiceOfLife/releases/tag/2.0.1-carrot from here but be warned that we encouter a false-positive in virus checking. I linked for this problem the jenkins download in https://github.com/GTNewHorizons/SpiceOfLife/issues/21 which I was be able to download.

Try one of these out (as I'm going to ask my players todo the same) and let us know if the crash still occours with this version :)

**Comment by Crystal98TR** [comment]
Sorry for late reply, it takes a while to open the pack.

Interestingly it didn't crash when I opened the Lunch Box but it still crashes when I try to open a quest with a food reward inside.

**Comment by Chaoschaot232** [comment]
> Sorry for late reply, it takes a while to open the pack.
> 
> Interestingly it didn't crash when I opened the Lunch Box but it still crashes when I try to open a quest with a food reward inside.

With both versions or with the 2.0.1 ones? Also attach any crash/error log even if the main content should be the same. Should help the Dev's to locate the problem :)

**Comment by Crystal98TR** [comment]
> With both versions or with the 2.0.1 ones?

Isn't it just one version, 2.0.1? I also can't download from the first link by the way, because I also use Bitdefender. Sorry if I'm dumb.

[crash-2022-05-04_04.22.17-client.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/8617470/crash-2022-05-04_04.22.17-client.txt)

**Comment by Zelda-Swordsman** [comment]
> Isn't it just one version, 2.0.1? I also can't download from the first link by the way, because I also use Bitdefender. Sorry if I'm dumb. 
> [crash-2022-05-04_04.22.17-client.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/8617470/crash-2022-05-04_04.22.17-client.txt)

The problem is Bitdefender deleted a class file from within the jar, specifically squeek/spiceoflife/shadow/com/udojava/evalex/Expression$Tokenizer.class

That's why deleting and restoring the file won't fix anything.

![image](https://user-images.githubusercontent.com/36285521/166687194-7a9157d4-62ea-4d4c-9c3a-a29029a3eb68.png)

You could try heading into Bitdefender and tell it to restore the file. My guess is some virus definitions were updated, causing the false positive. Trying to download the jar will just cause Bitdefender to delete the entire jar.

**Comment by Crystal98TR** [comment]
That makes a lot of sense, I don't know how I didn't think about that. I tried excluding it but it still doesn't work, probably because the file is just deleted. Then I tried to restore it but Bitdefender says it can't be restored. So I have to reinstall the mod. But no matter what I do or how many times I say it to exclude anything, it always deletes that file. Couldn't find a way around.

**Comment by glowredman** [comment]
@Crystal98TR Could you try [this](http://jenkins.usrv.eu:8080/job/SpiceOfLife/30/artifact/build/libs/SpiceOfLife-1.7.10-2.0.1-carrot-2-g49459ec.jar) version?

**Comment by Chaoschaot232** [comment]
> @Crystal98TR Could you try [this](http://jenkins.usrv.eu:8080/job/SpiceOfLife/30/artifact/build/libs/SpiceOfLife-1.7.10-2.0.1-carrot-2-g49459ec.jar) version?

When I try to klick your link within chrome the tab is imerdetaly closed and no download will start. Only copy and past the link in a new chrome tab works.

If somebody has the same problem, please go over http://jenkins.usrv.eu:8080/job/SpiceOfLife/ and use the 3rd file to download :)

In generel my MS Defender hasn't shown any interest of the new carrot file so it seems to be clean for now. My friend, who uses Bitdefender hasn't had any problem with the download and with starting the client. Also no crash report is thrown anymore so from our side there are no problems with the new file :)

**Comment by Crystal98TR** [comment]
> @Crystal98TR Could you try [this](http://jenkins.usrv.eu:8080/job/SpiceOfLife/30/artifact/build/libs/SpiceOfLife-1.7.10-2.0.1-carrot-2-g49459ec.jar) version?

Yo this one works! Thanks very much, but how though?

**Comment by glowredman** [comment]
> After this the jar shows up as clean on virustotal: https://www.virustotal.com/gui/url/d15fa979a666877f60176bd995fead8245fe82c67c8a5609bb7e6a53b125b4d5. Maven reports that the old version had a dependency with a confirmed CVE, maybe that was the reason it got flagged - who knows

See: https://github.com/GTNewHorizons/SpiceOfLife/pull/22
